internal/hasher/hasher.go

```diff
@@ -18,7 +18,10 @@ func GetShortLink(input string) (string, error) {
 
 	urlHashBytes := sha256Of(input)
 	generatedNumber := new(big.Int).SetBytes(urlHashBytes).Uint64()
-	finalString := base58Encoded([]byte(fmt.Sprintf("%d", generatedNumber)))
+	finalString, err := base58Encoded([]byte(fmt.Sprintf("%d", generatedNumber)))
+	if err != nil {
+		log.Fatal(err)
```

Comment:
```suggestion
		return "", err
```
Это как раз то, что я уже писал. Сервер не должен падать из-за сбоя какой-то внутренней функции. Правильное поведение - поднять ошибку максимально вверх (а это будет хендлер) и там ее обработать. И не упасть, а вернуть пользователю 500 (если это ошибка сервиса) или BadRequest, если эта ошибка связана с неверным вводом.